frost-p256/src/lib.rs

```diff
@@ -0,0 +1,277 @@
+    }
+
+    fn invert(scalar: &Self::Scalar) -> Result<Self::Scalar, Error> {
+        // [`p256::Scalar`]'s Eq/PartialEq does a constant-time comparison using
+        // `ConstantTimeEq`
+        if *scalar == <Self as Field>::zero() {
+            Err(Error::InvalidZeroScalar)
+        } else {
+            Ok(scalar.invert().unwrap())
+        }
+    }
+
+    fn random<R: RngCore + CryptoRng>(rng: &mut R) -> Self::Scalar {
+        Scalar::random(rng)
+    }
+
+    fn random_nonzero<R: RngCore + CryptoRng>(rng: &mut R) -> Self::Scalar {
+        loop {
+            let scalar = Scalar::random(&mut *rng);
+
+            // This impl of `Eq` calls to `ConstantTimeEq` under the hood
+            if scalar != Scalar::zero() {
+                return scalar;
+            }
+        }
+    }
+
+    fn serialize(scalar: &Self::Scalar) -> Self::Serialization {
+        scalar.to_bytes().into()
+    }
+
+    fn deserialize(buf: &Self::Serialization) -> Result<Self::Scalar, Error> {
+        let field_bytes: &p256::FieldBytes = buf.into();
+        match Scalar::from_repr(*field_bytes).into() {
+            Some(s) => Ok(s),
+            None => Err(Error::MalformedScalar),
+        }
+    }
+}
+
+#[derive(Clone, Copy, PartialEq)]
+/// An implementation of the FROST P-256 ciphersuite group.
+pub struct P256Group;
+
+impl Group for P256Group {
+    type Field = P256ScalarField;
+
+    type Element = ProjectivePoint;
+
+    /// [SEC 1][1] serialization of a compressed point in P-256 takes 33 bytes
+    /// (1-byte prefix and 32 bytes for the coordinate).
+    ///
+    /// Note that, in the P-256 spec, the identity is encoded as a single null byte;
+    /// but here we pad with zeroes. This is acceptable as the identity _should_ never
+    /// be serialized in FROST, else we error.
+    ///
+    /// [1]: https://secg.org/sec1-v2.pdf
+    type Serialization = [u8; 33];
+
+    fn order() -> <Self::Field as Field>::Scalar {
+        // TODO: rethink this, no way to represent the order in `Scalar`
+        Scalar::zero()
+    }
+
+    fn cofactor() -> <Self::Field as Field>::Scalar {
+        Scalar::one()
+    }
+
+    fn identity() -> Self::Element {
+        ProjectivePoint::IDENTITY
+    }
+
+    fn generator() -> Self::Element {
+        ProjectivePoint::GENERATOR
+    }
+
+    fn serialize(element: &Self::Element) -> Self::Serialization {
+        let mut fixed_serialized = [0; 33];
+        let serialized_point = element.to_affine().to_encoded_point(true);
+        let serialized = serialized_point.as_bytes();
+        // Sanity check; either it takes all bytes or a single byte (identity).
+        assert!(serialized.len() == fixed_serialized.len() || serialized.len() == 1);
+        {
+            // Copy to the left of the buffer (i.e. pad the identity with zeroes).
+            let (left, _right) = fixed_serialized.split_at_mut(serialized.len());
+            left.copy_from_slice(serialized);
+        }
+        fixed_serialized
+    }
```

Comment:
The API for `serialize()` currently always ensures success (as per the FROST spec), but, we have been discussing making the API fallible in case someone tries to serialize() the identity (which should fail with an Error).

This is not in the FROST spec yet and not in the `frost-core` API we're currently implementing, so I'm fine having this as-is but with the forward-looking expectation that `serialize()` will, when updated to implement the applicable version of the FROST spec, become `serialize() -> Result<Self::Serialization, Error>`, similar to `deserialize()`, and instead of panic'ing on the failed assert, return `Err(Error)`